src/app/Shared/Services/Target.service.tsx

```diff
@@ -51,7 +51,11 @@ class TargetService {
   private readonly _credentials: Map<string, string> = new window.Map();
 
   setTarget(target: Target): void {
-    this._target.next(target);
+    if (target === NO_TARGET || (target.alias != '' && target.connectUrl != '')) {
```

Comment:
If I do `setTarget({ alias: 'something', connectUrl: undefined} as Target)`, does this throw an error or accept the state change? What should it do?

Reply:
You are right, I changed it to !!target.alias, it seems that checks for null, undefined and empty string

Reply:
Pretty much.

https://developer.mozilla.org/en-US/docs/Glossary/Falsy

https://medium.com/better-programming/javascript-bang-bang-i-shot-you-down-use-of-double-bangs-in-javascript-7c9d94446054